### ISSUE TITLE
train: codingjepa.training.manager (Lightning module + Manager analog)

### ISSUE THREAD

**Issue by AbdelStark** [AbdelStark/CodingJEPA#67]
## Context

RFC-0008 §D1: keep an option to re-implement a thin slice of `stable_pretraining` rather than carrying it as a hard dep. The manager exposes the LeWM-style `forward` callback and a minimal training driver.

## Scope

- `codingjepa.training.manager.Manager` — orchestrates dataloader iteration, optimizer step, AMP autocast (bf16), gradient clipping (1.0), grad accumulation, scheduler step, callbacks, checkpoint writing.
- Reads the resolved Hydra config; dumps it to `runs/<run_id>/config.yaml`.
- Records `seed`, `git_sha`, `manifest_hash`, `tokenizer_hash` in the run dir.

## Out of Scope

Specific callbacks (separate issues).

## Design Reference

- RFC: [RFC-0008 §D1 §D9](../blob/main/docs/rfcs/RFC-0008-training-recipe-and-pipeline.md)
- SPEC: [`docs/spec/05-observability.md`](../blob/main/docs/spec/05-observability.md) §Per-run directories

## Acceptance Criteria

- [ ] Trains 1k steps on a tiny fixture without errors.
- [ ] Resolved config + seeds.json + env.json written.

## Dependencies

- Depends on: #66, #17

## Metadata

```
type: feature
area: training
priority: p0
effort: l
spec: rfc-0008
milestone: v0.2
tracking: #6
```
